### ISSUE TITLE
replicating results

### ISSUE THREAD

**Issue by jpilaul** [YiSyuanChen/MTL-ABS#4]
Hi, thank you so much for making your code public. Quick question on getting the `../models/pre_train/pre_ext_abs_cnndm/model_step_148000.pt` model for `-train_from` argument. Can you give more details? Did you just pretrain the model on cnndm abstractive summarization?
Thanks!

**Comment by jpilaul** [comment]
I am guessing the argument is important. When I remove it, I get the following validation performance and I am not able to replicate results:

```
[2021-07-22 18:42:15,968 INFO] Rouges at step 2
>> ROUGE-F(1/2/3/l): 0.00/0.00/0.00
ROUGE-R(1/2/3/l): 0.00/0.00/0.00
```

**Comment by YiSyuanChen** [comment]
Hi, thanks for your interest in our work. Yes, the model is just pre-trained on CNNDM dataset, and  `-train_from` is the path for this pre-trained model. Our base model shares the same architecture with [PreSumm](https://github.com/nlpyang/PreSumm). You could directly download the [pre-trained model ](https://drive.google.com/file/d/1-IKVCtc4Q-BdZpjXc4s70_fRsWnjtYLr/view)they provided in the repository.

**Comment by jpilaul** [comment]
Thanks for sharing the link to the model. I have reran the experiments and I still cannot replicate the paper results for aeslc. The paper mentions "the whole model is trained with 6K meta-steps" but in the readme has `--train_steps 2`. Should I set `--train_steps 6000`? Currently with hyperparameters in the repo, I am getting:
```
[2021-07-23 15:17:15,742 INFO] Rouges at step 2 
>> ROUGE-F(1/2/3/l): 12.05/4.26/11.14
ROUGE-R(1/2/3/l): 35.72/14.40/33.28
```

**Comment by jpilaul** [comment]
I tried `--train_steps 6000` and the best score that I have gotten was:
```
[2021-07-27 13:25:34,360 INFO] Rouges at step 4000 
>> ROUGE-F(1/2/3/l): 12.77/5.04/12.39
ROUGE-R(1/2/3/l): 16.56/6.65/16.12
```
The paper reports ` ROUGE-F(1/2/3/l): 21.27/10.79/20.85 ` which is twice as a high...
I am unable to replicate results.

**Comment by YiSyuanChen** [comment]
Hi, I want to check what kinds of meta-datasets you used for training?

**Comment by jpilaul** [comment]
Hi, thanks for your reply. I am using reddit, reddit_tifu  and wikihow for AESLC.

**Comment by YiSyuanChen** [comment]
Hi, I have reproduced the results of AESLC, and the ROUGE scores are:
```
[2021-07-28 12:03:26,032 INFO] Rouges at step 16 
>> ROUGE-F(1/2/3/l): 21.45/10.56/21.01
ROUGE-R(1/2/3/l): 23.82/11.81/23.35
```
Here are my procedures:
1. Perform training for meta-transfer learing. To make quick check, I reduce the train_steps from 6000 to 4000.
2. Perform validation for meta-transfer learining. Again, I reduce the train_steps from 600 to 100 for quick check. I got my best meta-trained model at step 4000 (in loss).
3. Perform training for adaptation with **10 AESLC data** and the **best meta-trained model**.
4. Perform validation for adaptation. I got my **best meta-trained and adapted model** at step 16 (in ROUGE). 
(NOTE: the best ROUGE scores may not always happen at the lowest loss, the code will use the top-5 model to decode.)
5. Perform test for adaptation with the **best meta-trained and adapted model**. I then got the results shown above.

I guess you get the results with the **best meta-trained model** at second step, where the model has not been adapted with few target domain data. I also try to test with the best meta-trained model, and the ROUGE scores are:
```
[2021-07-28 12:10:31,115 INFO] Rouges at step 4000 
>> ROUGE-F(1/2/3/l): 12.22/4.97/11.79
ROUGE-R(1/2/3/l): 19.00/7.95/18.34
```
which is pretty close to your results. The meta-learning process may not prodcue model that is good for zero-shot setting, since the purpose of training is to adapt with few target examples. 

You may also download my [reproduce results](https://drive.google.com/drive/folders/1XaL3-XuAXphMYcEZJ2Ls0nWDolS0NY0D?usp=sharing), it includes:
1. Procedures as a script
2. AESLC dataset with 10 training examples and full validation/test examples
3. The best meta-trained model at step 4000
4. The best meta-trained and adapted model at step 16

To make a quick try, you could directly test the best meta-trained and adapted model. Hope these informations are helpful for you :)

**Comment by iin729** [comment]
good job and perfect answer
